Content.Server/Store/Systems/StoreSystem.Ui.cs

```diff
@@ -154,20 +176,84 @@ private void OnBuyRequest(EntityUid uid, StoreComponent component, StoreBuyListi
         foreach (var currency in listing.Cost)
         {
             component.Balance[currency.Key] -= currency.Value;
+
+            if (!component.BalanceSpent.ContainsKey(currency.Key))
+                component.BalanceSpent.Add(currency.Key, FixedPoint2.Zero);
+
+            component.BalanceSpent[currency.Key] += currency.Value;
```

Comment:
Use TryGetValue, if you really wanted to get fancy collectionsmarshal exists with an even better version but it's probably too based for content.